extensions/objects/templates/ImageWithRegionsEditor.js

```diff
@@ -274,13 +339,77 @@ oppia.directive('imageWithRegionsEditor', [
             $scope.hoveredRegion = index;
           }
         };
+        $scope.onMouseMoveRegion = function() {
+          if ($scope.userIsCurrentlyDragging ||
+              $scope.userIsCurrentlyResizing) {
+            return;
+          }
+          region = reg = cornerAndDimensionsFromRegionArea(
+            $scope.$parent.value.labeledRegions[
+              $scope.hoveredRegion].region.area);
+          if (!$scope.xDirectionToggled && !$scope.yDirectionToggled) {
+            if ($scope.mouseY <= region.y + $scope.resizeRegionBorder &&
+                $scope.mouseX >= region.width + region.x -
+                $scope.resizeRegionBorder) {
+              // Along top right
+              $scope.xDirection = -1;
+              $scope.yDirection = 1;
+            } else if ($scope.mouseY <= region.y + $scope.resizeRegionBorder &&
+                $scope.mouseX <= region.x + $scope.resizeRegionBorder) {
+              // Along top left
+              $scope.xDirection = 1;
+              $scope.yDirection = 1;
+            } else if ($scope.mouseY >= region.height + region.y -
+                $scope.resizeRegionBorder && $scope.mouseX >= region.width +
+                region.x - $scope.resizeRegionBorder) {
+              // Along bottom right
+              $scope.xDirection = -1;
+              $scope.yDirection = -1;
+            } else if ($scope.mouseY >= region.height + region.y -
+                $scope.resizeRegionBorder && $scope.mouseX <= region.x +
+                $scope.resizeRegionBorder) {
+              // Along bottom left
+              $scope.xDirection = 1;
+              $scope.yDirection = -1;
+            } else if ($scope.mouseY <= region.y + $scope.resizeRegionBorder) {
+              // Along top
+              $scope.yDirection = 1;
+              $scope.xDirection = 0;
```

Comment:
A tip for the future: don't swap xDirection and yDirection subtly like this. Keep each branch looking "the same" (though hopefully we can still reduce some of the branching -- see suggestion above).